src/components/entity/list.vue

```diff
@@ -458,6 +498,105 @@ export default {
       // less than the lowest size option, hence we don't need to make a request.
       if (this.odataEntities.count < this.pageSizeOptions[0]) return;
       this.fetchChunk(false);
+    },
+    clearSelectedEntities() {
+      this.odataEntities.value.forEach(e => { e.__system.selected = false; });
+      this.selectedEntities.clear();
+    },
+    requestBulkDelete() {
+      const uuids = Array.from(this.selectedEntities);
+      // TODO: disable the whole table
+
+      const bulkDelete = () => this.request({
+        method: 'POST',
+        url: apiPaths.entities(this.projectId, this.datasetName, '/bulk-delete'),
+        alert: false,
+        data: {
+          ids: uuids
+        }
+      });
+
+      const onSuccess = () => {
+        if (this.deletedEntityCount.dataExists) this.deletedEntityCount.value += uuids.length;
+
+        uuids.forEach(uuid => this.odataEntities.removedEntities.add(uuid));
+
+        this.deletedEntities = [];
+        this.deletedEntities.push(...this.odataEntities.value.filter(e => uuids.includes(e.__id)));
+        this.odataEntities.value = this.odataEntities.value.filter(e => !uuids.includes(e.__id));
+        this.alert.success(this.$tcn('alert.bulkDelete', this.selectedEntities.size))
+          .cta(this.$t('undo'), () => this.requestBulkRestore(uuids));
```

Comment:
I'm slightly concerned about the user doing something in the table (like changing a filter) right after the bulk delete, and only _then_ clicking Undo. I'm not as concerned about it as some of the other cases though, since the toast will automatically disappear along with the Undo button after a short amount of time. The only problem cases I can think of are:

1. Change a filter, then click Undo. The deleted entities will reappear even if the new filter doesn't apply to them.
2. View deleted entities, then click Undo. This one seems a little more likely (e.g., someone has second thoughts after viewing the trash). But again, it'd have to happen within 7 seconds.

Do you think we should make any changes in this area? Maybe this one is too much of an edge case?